packages/amplify-category-function/src/provider-utils/awscloudformation/utils/truncateResourceNames.ts

```diff
@@ -0,0 +1,53 @@
+import { v5 as uuid } from 'uuid';
+
+const funcNameMaxLen = 64; // max length allowed by Lambda service
+const roleNameMaxLen = 64; // max length allowed by IAM service
+
+type Params = {
+  functionName?: string;
+  roleName?: string;
+};
+
+export const truncateResourceNames = (params: Readonly<Params>): Params => {
```

Comment:
+1 on using `Readonly`, not sure if this should be enforced, but we should at least try to make our function parameters immutable as a default approach